### PR TITLE
add new module vt/decompose_blocksub

### PR DESCRIPTION
**Pull request by kubranarci**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Review by LouisLeNezet — CHANGES_REQUESTED** [review]
Just a few tweaks but good to go for me 😉
Could I ask you to review the following in exchange ?
https://github.com/nf-core/test-datasets/pull/1407

File: modules/nf-core/vt/decomposeblocksub/main.nf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    label 'process_low'
+
+    // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/vt:2015.11.10--h5ef6573_4':
+        'biocontainers/vt:2015.11.10--h5ef6573_4' }"
+
+    input:
+    tuple val(meta), path(vcf), path(intervals)
```
[the remark this answers is not in the document]
Comment (kubranarci) [reply modules/nf-core/vt/decomposeblocksub/main.nf:12]: it works both with vcf and vcf.gz, if index is not provided it creates.

File: modules/nf-core/vt/decomposeblocksub/main.nf [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    def VERSION = "2015.11.10" // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+
+    """
+    vt decompose_blocksub \\
+        -o ${prefix}.vcf \\
+        ${bed} \\
+        ${args} \\
+        ${vcf}
+
+    bgzip ${args2} --threads ${task.cpus} ${prefix}.vcf
```
[the remark this answers is not in the document]
Comment (kubranarci) [reply modules/nf-core/vt/decomposeblocksub/main.nf:40]: I realized output can directly be vcf.gz 👯‍♀️

File: modules/nf-core/vt/decomposeblocksub/main.nf [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    def args = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    if ("$vcf" == "${prefix}.vcf" || "$vcf" == "${prefix}.vcf.gz") {
+        error "Input and output names are the same, set prefix in module configuration to disambiguate!"
+    }
+
+    def bed = intervals ? "-i ${intervals}" : ""
+    def VERSION = "2015.11.10" // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
```
[the remark this answers is not in the document]
Comment (kubranarci) [reply modules/nf-core/vt/decomposeblocksub/main.nf]: version is being extracted automatically now.

File: modules/nf-core/vt/decomposeblocksub/meta.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  - vcf:
+      - meta:
+          type: map
+          description: |
+            Groovy Map containing sample information
+            e.g. `[ id:'test', single_end:false ]`
+      - "*.vcf.gz":
+          type: file
+          description: The decomposed VCF file
+          pattern: "*.vcf.gz"
```
[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply modules/nf-core/vt/decomposeblocksub/meta.yml:42]: Here is an example:
https://github.com/nf-core/modules/blob/master/modules/nf-core/coptr/index/meta.yml

**Review by LouisLeNezet — CHANGES_REQUESTED** [review]

File: modules/nf-core/vt/decomposeblocksub/meta.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (kubranarci) [reply modules/nf-core/vt/decomposeblocksub/meta.yml:52]: thank you, I have added links now

File: modules/nf-core/vt/decomposeblocksub/main.nf (hunk 1 above)
[the remark this answers is not in the document]
Comment (kubranarci) [reply modules/nf-core/vt/decomposeblocksub/main.nf]: I implemented the suggestion

File: modules/nf-core/vt/decomposeblocksub/main.nf [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    if ("$vcf" == "${prefix}.vcf" || "$vcf" == "${prefix}.vcf.gz") {
```
Comment (LouisLeNezet) [inline modules/nf-core/vt/decomposeblocksub/main.nf:26]: The vcf check is not necessary as you force it to compressed format.
```suggestion
    if ("$vcf" == "${prefix}.vcf.gz") {
```

File: modules/nf-core/vt/decomposeblocksub/main.nf [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        vt: \$(echo \$(vt decompose_blocksub 2>&1 | grep 'decompose_blocksub v' | sed -n 's/.*decompose_blocksub v\\([0-9]\\+\\.[0-9]\\+\\).*/\\1/p' ))
+    END_VERSIONS
+    """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    if ("$vcf" == "${prefix}.vcf" || "$vcf" == "${prefix}.vcf.gz") {
```
Comment (LouisLeNezet) [inline modules/nf-core/vt/decomposeblocksub/main.nf:47]:
```suggestion
    if ("$vcf" == "${prefix}.vcf.gz") {
```

File: modules/nf-core/vt/decomposeblocksub/tests/main.nf.test [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    tag "vt"
+    tag "vt/decomposeblocksub"
+
+    test("test-vt-decomposeblocksub") {
+
+        when {
+            process {
+                """
+                input[0] = [
+				[ id:'test', single_end:false ], // meta map
```
Comment (LouisLeNezet) [inline modules/nf-core/vt/decomposeblocksub/tests/main.nf.test:20]: Could you add an indentation here ? and for the other lines inside `input[0]` in the 2 tests ?
```suggestion
				    [ id:'test', single_end:false ], // meta map
```

**Review by LouisLeNezet — APPROVED** [review]
Just a few tweaks but good to go 😉

File: modules/nf-core/vt/decomposeblocksub/main.nf (hunk 5 above)
Comment (kubranarci) [reply modules/nf-core/vt/decomposeblocksub/main.nf:26]: input file can also be vcf (both is accepted by the tool)

**Pull request merged**